Sources/CoreMetrics/Metrics.swift

```diff
@@ -488,6 +580,76 @@ public protocol MetricsFactory {
     func destroyTimer(_ handler: TimerHandler)
 }
 
+/// Wraps a CounterHandler, adding support for incrementing by floating point values by storing an accumulated floating point value and recoding increments to the underlying CounterHandler after crossing integer boundaries.
+private class AccumulatingRoundingFloatingPointCounter: FloatingPointCounterHandler {
+    private let lock = Lock()
+    private var fraction: Double = 0
+    private let counterHandler: CounterHandler
+
+    init(counterHandler: CounterHandler) {
+        self.counterHandler = counterHandler
+    }
+
+    func increment(by amount: Double) {
+        // FIXME: what do?
+        let maxIncrement = Double(Int64.max - (2 << 8))
+        assert((0...maxIncrement).contains(amount))
+        let amount = min(amount, maxIncrement)
```

Comment:
not exactly sure I understand what the above is validating, its it that amount < Int64.max?

Reply:
I wanted to make sure that doubles passed to this integer rounded counter were representable, the newer version would look like this:
```swift
        let largestRepresentableInt64InFloat64 = Double(2 << 53)
        assert((0...largestRepresentableInt64InFloat64).contains(amount))
        let amount = min(amount, largestRepresentableInt64InFloat64)
```

I'm not sure if this is the right behavior though.

Reply:
This behavior changed, we should probably start a new thread